src/main/java/org/mule/extension/http/internal/request/HttpResponseToResult.java

```diff
@@ -51,23 +52,14 @@
 
   private static final Logger logger = LoggerFactory.getLogger(HttpResponseToResult.class);
 
-  private static final String BINARY_CONTENT_TYPE = BINARY.toRfcString();
+  static final String BINARY_CONTENT_TYPE = BINARY.toRfcString();
   private static boolean STRICT_CONTENT_TYPE = parseBoolean(getProperty(SYSTEM_PROPERTY_PREFIX + "strictContentType"));
 
   private final Function<String, MediaType> parseMediaType = memoize(ctv -> parseMediaType(ctv), new ConcurrentHashMap<>());
 
-  public Result<InputStream, HttpResponseAttributes> convert(HttpRequesterCookieConfig config, MuleContext muleContext,
-                                                             HttpResponse response, URI uri) {
-    String responseContentType = response.getHeaderValue(CONTENT_TYPE);
-
-    HttpEntity entity = response.getEntity();
-
-    if (isEmpty(responseContentType) && !empty(entity)) {
-      // RFC-2616 specifies application/octet-stream as default when none is received
-      responseContentType = BINARY_CONTENT_TYPE;
-    }
-
-    MediaType responseMediaType = getMediaType(responseContentType, getDefaultEncoding(muleContext));
+  Result<InputStream, HttpResponseAttributes> convert(HttpRequesterCookieConfig config, MuleContext muleContext,
+                                                      HttpResponse response, HttpEntity entity,
+                                                      CursorProvider payloadCursorProvider, URI uri) {
```

Comment:
can you pass a `Supplier<InputStream>` instead of coupling this to `CursorProvider`?

Reply:
Done!